x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/TokenizerUtils.java

```diff
@@ -39,7 +39,11 @@ static LinkedList<DelimitedToken> splitOutNeverSplit(CharSequence input, CharTri
                 if (neverSplitSet.contains(maybeNeverSplit)) {
                     if (windowStart < neverSplitStart) {
                         bigTokens.add(
-                            new DelimitedToken(new CharSequenceRef(input, windowStart, neverSplitStart), windowStart, neverSplitStart)
+                            new DelimitedToken(
+                                new CharSequenceRef(input, windowStart, neverSplitStart - windowStart),
```

Comment:
`CharSequenceRef` parameters are:

 - CharSequence
 - offset
 - LENGTH

Previously it pointed to the last index of the nextsplit, not the length 🤦